lib/Sema/CSSimplify.cpp

```diff
@@ -4092,11 +4092,22 @@ bool ConstraintSystem::repairFailures(
     path.pop_back();
     // If this is a contextual mismatch between l-value types e.g.
     // `@lvalue String vs. @lvalue Int`, let's pretend that it's okay.
-    if (!path.empty() && path.back().is<LocatorPathElt::ContextualType>()) {
-      auto *locator = getConstraintLocator(anchor, path.back());
-      conversionsOrFixes.push_back(
-          IgnoreContextualType::create(*this, lhs, rhs, locator));
-      break;
+    if (!path.empty()) {
+      if (path.back().is<LocatorPathElt::ContextualType>()) {
+        auto *locator = getConstraintLocator(anchor, path.back());
+        conversionsOrFixes.push_back(
+            IgnoreContextualType::create(*this, lhs, rhs, locator));
+        break;
+      }
+
+      // If this is a function type param type mismatch in any position,
+      // the mismatch we want to report is for the whole structural type.
+      while (path.back().is<LocatorPathElt::TupleElement>()) {
```

Comment:
Let's use an alternative approach and preserve `path` here for the fallthrough if it's not at function argument position - use rfind to determine where `FunctionArgument` is and then re-create `path` like `{path.begin(), path.begin() + <function arg position>}` or use `drop_back(size_t N)`?

Reply:
Makes sense, I just fixed =]